Modules/Segmentation/LevelSets/include/itkParallelSparseFieldLevelSetImageFilter.h

```diff
@@ -124,16 +124,16 @@ class ITK_TEMPLATE_EXPORT ParallelSparseFieldCityBlockNeighborList
   Print(std::ostream & os, Indent indent) const;
 
 private:
-  char                      m_Pad1[128];
-  unsigned int              m_Size;
-  RadiusType                m_Radius;
-  std::vector<unsigned int> m_ArrayIndex;
-  std::vector<OffsetType>   m_NeighborhoodOffset;
+  char                      m_Pad1[128]{};
```

Comment:
@jhlegarreta @dzenanz Please wait, before merging. I'm still considering to propose adjusting the ITK Coding Style Guide, to exclude padding bytes from `{}` initialization.

Reply:
Maybe we could make it clearer that these are just padding bytes that do not need to be initialized by renaming `m_Pad1` and `m_Pad2` to something like `m_UninitializedPaddingBytes1` and  `m_UninitializedPaddingBytes2`...? We could also possibly make it clearer by declaring them as arrays of `byte`, instead of arrays of `char`. C++17 defines an `std::byte` type that we can easily borrow: https://en.cppreference.com/w/cpp/types/byte

Reply:
@N-Dekker   I agree with our last comment here.  the m_Pad1 is not really a state variable of the class.  It is a solution to a problem that happens to use a char array to enforce performance gains, but does not represent the state of the class.

These pad variables should have good documentation that they are excluded from the "standard rules" applied to member variables that represent the state of the object.

Reply:
When did we move to C++14? Maybe it is time to move to C++17?

Reply:
- @dzenanz Check pull request #2563
C++17 has got some really great improvements. 😃

Reply:
That was almost 2 years ago. I suggest making a forum topic about move to C++17, to see what others think about it. Perhaps describe implications on compiler support, and other similar considerations in the post.

Reply:
This PR has other improvements which are not contentious. I will merge this. @N-Dekker can then make a follow-up PR which just deals with Padding bytes.